keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -208,6 +208,18 @@ class SqlArtifactRepository(
         .value1()
     } > 0
 
```

Comment:
You can't do this this way - there are potentially many artifacts that will have the same name and type. You can only look up a delivery artifact given the artifact config.

For example:
Let's say several teams are deploying redis from the docker image `nflx/redis` provided by a centralized team. Each team is running their own redis cluster from that image, and wants to update it when a new version is rolled out. Each team has a redis artifact defined in their config.

Since there are several artifacts referring to redis by `name = nflx/redis` and `type = docker`, you can't find the delivery artifact by just knowing the name and type. You have to know the delivery config before you can find a delivery artifact.

Reply:
I see. I didn't really change the implementation in the caller function tho - we used to check if an artifact is registered based on name + type, and this function is just returning one of them. I don't understand why it's ok to use this logic in the `isRegistered` and not to fetch the actual artifact (although I understand your example).